File: frontend/src/microservices/order/objects/CreatePaymentLinkPlannerRequest.ts
// 本文件定义支付链接创建请求，和后端 `CreatePaymentLinkPlannerRequest` 保持同名镜像。

export type CreatePaymentLinkPlannerRequest = {
  orderId: string
  userId: string
  paymentMethod: string
  language: string | null
  publicBackendOrigin: string | null
}

export const createPaymentLinkPlannerRequestFromJson = (json: string): CreatePaymentLinkPlannerRequest =>
  JSON.parse(json) as CreatePaymentLinkPlannerRequest

export const createPaymentLinkPlannerRequestToJson = (value: CreatePaymentLinkPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/order/objects/CreatePaymentLinkPlannerRequest.scala
package com.typesafe.travel.order.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreatePaymentLinkPlannerRequest(orderId: String, userId: String, paymentMethod: String, language: Option[String], publicBackendOrigin: Option[String])
object CreatePaymentLinkPlannerRequest:
  given Encoder[CreatePaymentLinkPlannerRequest] = deriveEncoder
  given Decoder[CreatePaymentLinkPlannerRequest] = deriveDecoder
